projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts

```diff
@@ -187,22 +198,30 @@ export function selectDeliveryMethod() {
     'GET',
     '/rest/v2/electronics-spa/cms/pages?*/checkout/payment-details*'
   ).as('getPaymentPage');
-  cy.get('.cx-checkout-title').should('contain', 'Shipping Method');
+  cy.get('cx-checkout-progress a.is-active').should(
+    'contain',
+    '2. Delivery mode'
```

Comment:
Isn't this the same behaviour from what we had from the `first review`, where it failed in jenkins because it couldn't find this selector after clicking continue. The added assertions in `selectPaymentMethod()` will never happen if this failed.

https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/6860/commits/4b36a0f205c7db81d284b9f6e9cf50dfffa111d2
----------

I just did a deep dive just now, and I was thinking maybe it failed because it didn't finish loading the cart after entering the shipping address page, but it clicked continue already (MAYBE). 

Have you tried putting a wait on `/users/current/carts/00000011?fields=deliveryAddress(FULL),deliveryMode,paymentInfo(FULL)&lang=en&curr=USD` request. 

Looking at when the user enters the checkout shipping step, the above was the last request (configure the request to your needs). The flow would be wait for this checkout shipping page (already done) and after wait for the above request (which will ensure the cart to be 100% ready). Afterwards, move on to the clicking continue to the delivery page. I'm 30% confident the above will fix it, but at least there's hope this. If not, lets dive deeper on this or we might have to hack it somehow